dev/com.ibm.ws.jaxrs.2.0.client/src/com/ibm/ws/jaxrs20/client/security/LibertyJaxRsClientSSLOutInterceptor.java

```diff
@@ -156,21 +185,55 @@ public Method run() throws NoSuchMethodException, SecurityException {
                 }
             });
 
-            String uriString = (String) message.get(Message.REQUEST_URI);
-            URI uri = URI.create(uriString);
-            int port = uri.getPort();
+            SSLSocketFactory ssLSocketFactory = (SSLSocketFactory) m.invoke(classObject, getJaxRsSSLManagerParams(sslRef, message));
+            return ssLSocketFactory;
+        } catch (Exception e) {
+            if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+                Tr.debug(tc, "getSSLSocketFactoryBySSLRef reflection failed with exception " + e.toString());
+            }
+            return null;
+        }
+    }
 
-            // if the port wasn't specified, use the default SSL port (443)
-            if (port == -1 && !uriString.contains(":-1")) {
-                port = 443;
+    private List<String> getSSLCipherSuites(String sslRef, Message message) {
+        try {
+            final Class<?> jaxrsSslMgrClass = getJaxRsSSLManagerClass();
+            Object classObject = jaxrsSslMgrClass.newInstance();
+            Method m = AccessController.doPrivileged(new PrivilegedExceptionAction<Method>() {
+
+                @Override
+                public Method run() throws NoSuchMethodException, SecurityException {
+                    return jaxrsSslMgrClass.getDeclaredMethod("getSSLCipherSuitesBySSLRef", String.class, String.class, String.class);
+                }
+            });
+
+            String[] cipherSuites = (String[]) m.invoke(classObject, getJaxRsSSLManagerParams(sslRef, message));
+            return Arrays.asList(cipherSuites);
+        } catch (Exception e) {
+            if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+                Tr.debug(tc, "getSSLCipherSuitesBySSLRef reflection failed with exception " + e.toString());
             }
+            return null;
+        }
+    }
 
-            Object[] parameters = { sslRef, uri.getHost(), Integer.toString(port) };
-            SSLSocketFactory ssLSocketFactory = (SSLSocketFactory) m.invoke(classObject, parameters);
-            return ssLSocketFactory;
+    private List<String> getSSLProtocols(String sslRef, Message message) {
+        try {
+            final Class<?> jaxrsSslMgrClass = getJaxRsSSLManagerClass();
+            Object classObject = jaxrsSslMgrClass.newInstance();
+            Method m = AccessController.doPrivileged(new PrivilegedExceptionAction<Method>() {
```

Comment:
Question:  Is this doPriv needed for Java 2 security only?   If so it should be gated by an if check.

Reply:
dunno, this is how we get the SSLSocketFactory already so I'm just mirroring that. I'm hesitant to make any changes.